### ISSUE TITLE
Add option to only track events from specific hostname(s)

### ISSUE THREAD

**Issue by tim-the-arcane** [umami-software/umami#2328]
### Describe the feature or enhancement

It seems that Umami tracks any event on whatever hostname (production, staging, localhost) as long as its script is loaded and the `data-website-id` attribute points to the page ID that was setup in the dashboard. 

I tested this by using the script tag in a local preview and also a deployed version on Vercel on a barebones HTML file with the script tag in the `<head>`. Both hostnames obviously deviate from the domain in the settings:

![image](https://github.com/umami-software/umami/assets/10296479/0758d3b8-784b-442f-a0f5-c84d48f7519c)

This certainly helps when first implementing Umami tracking in a new website and testing if events come through. But after that the statistics could be potentially flooded by unwanted page views and events for various reasons:

1. local development environments
2. staging deployments
3. bad actors who want to skew the collected data of a website

I know that you can set the domains to track by explicitly adding them to the `data-domains` attribute on the script tag. But this only helps for the first 2 issues. Anyone who would want to destroy the real usage statistics of a site could just create a simple HTML file, add the script tag without the `data-domains` attribute, host it locally and slam it with requests.

So I propose to add an option in the websites settings to only record events from authorized refs on the side of Umami and drop requests from other sources. This could be either a simple checkbox/switch or a list of eligible hostnames (although I currently see no value in tracking multiple hostnames into one website anyway, except for language subdomains maybe...)

**Comment by FHU-yezi** [comment]
But bad actors can simulate an referer field that makes Umami think it is from the right source.

**Comment by tim-the-arcane** [comment]
@FHU-yezi Thanks for your answer! 

I checked this and you're totally right. I entirely missed this. So it's impossible to completely prevent this kind of attack from the client-side. All the other Analytics tools would have the same issue then and only seem more secure to this by adding such options in their settings. This requested feature would mostly be feature creep then.

Just out of curiosity I will look into tracking events server-side - for example with a Next.js middleware. Maybe I can cook something up and share it with the community.

Thanks again for the clarification, I learned a lot about HTTP headers by researching this further! Will clean up behind me and close this issue now. 🙈
